rfcs/text/0000-data-processing-robustness.md

```diff
@@ -0,0 +1,73 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC DCP data processing robustness principles
+
+## Summary
+
+This document describes requirements for individual DCP components related to robustness against incorrect, corrupt, or unexpected data.  It also lays out expectations for data integrity and error handling for DCP software.  It does not specify implementation details or technology.
+
+## Author(s)
+
+[Mark Diekhans](mailto:markd@ucsc.edu),
+[Tony Burdett](mailto:tburdett@ebi.ac.uk)
+
+## Shepherd
+[Tony Burdett](mailto:tburdett@ebi.ac.uk)
+
+## Motivation
+
+The HCA DCP is driven by input generated from external sources. As most developers on biological data processing systems can attest, this produces data that is highly variable and error-prone. Collecting, storing and processing this type of content can pose challenges for software, as systems must be highly robust against all possible types of problematic data. When unexpected content is encountered, it needs to be handled reliably and predictably.
```

Comment:
It's difficult to handle something _predictably_ when that something is _unexpected_. But I think this is semantics and likely we want all things that don't fall into the bin of 'not the norm' to be handled reliably and predictably.